### PR TITLE
Improve rxxp indices

### LINKED ISSUE #931: Feature request: Provide the ECAD definition of R75p R95p and R99p indices.
* xclim version: 0.31.x
* Python version: all
* Operating System: all

### Description

Currently, the indices R75p R95p and R99p (as per icclim.yml file) rely on `days_over_precip_thresh` function which expect in `per` parameter a DataArray of **doy** percentiles.

However, when reading the definition from [ECAD](https://www.ecad.eu/documents/atbd.pdf#subsubsection.5.3.8) documentation:
> R75p
• Days with RR > 75th percentile of daily amounts (moderate wet days)
(days)
Let RRwj be the daily precipitation amount at wet day w (RR ≥ 1.0 mm)
of period j and let RRwn75 be the 75th percentile of precipitation at wet
days in the 1961–1990 period. Then counted is the number of days where:
RRwj > RRwn75

As per this definition, the percentiles should be computed on the whole base period (above 61-90) and not for each day of year.

Thus, I would like to implement an index function similar to `days_over_precip_thresh` with a `thresh` DataArray parameter having values thresholds for each spatial point.
Or maybe simply replace `days_over_precip_thresh` but both definition could make sens I suppose.
icclim.yml should be updated to probably follow ECAD definition as icclim v4 does.

### What I Did

I compared the output of icclimv4 and found very different results.

Notes: 
- **I'm willing to work on this issue.**
- climdex/climpact use again another definition for r99p (and others indices). It sum the precipitation amounts above the percentile thresholds and the percentiles are **not** for doy.
- The subject of RxxP definition is discussed on clix-meta: https://github.com/clix-meta/clix-meta/issues/53.
- The [WMO definition](https://library.wmo.int/doc_num.php?explnum_id=9419) for R95p is 
>R95p [...] the number of days with rainfall above the 95th percentile of daily accumulations.

Thus current `days_over_precip_thresh` is following a valid definition, the one from WMO, but it differ from icclim which uses ECAD definition.

### PR DESCRIPTION
**Pull request by bzah**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #931
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.
- [ ] `bumpversion patch` has been called on this branch
- [x] The relevant author information has been added to `.zenodo.json`

### What kind of change does this PR introduce?
- Add new indicator `days_over_precip_doy_thresh` for the WMO definition of Rxxp indices.
The existing `days_over_precip_thresh` (notice ~doy~) description is now made for the ECA&D definiton of Rxxp indices.
Note that both indicators call the same **indice** function, only the description is changed.
- Also added some error handling when non doy percentiles are used in BS.

### Does this PR introduce a breaking change?
Not a breaking change, but those who calls `days_over_precip_thresh` with `per` computed using doy values will receive an improper description for the indicator. 
They should start using `days_over_precip_doy_thresh` for this purpose.

### Other information:
ECAD (pdf): https://www.ecad.eu/documents/atbd.pdf
WMO: https://library.wmo.int/index.php?lvl=notice_display&id=138#.YUBV6CVS9ZI

### CODE REVIEW

File: xclim/indicators/atmos/_precip.py [hunk 1]
```diff
@@ -290,13 +290,22 @@ class HrPrecip(Hourly):
...
     standard_name="number_of_days_with_lwe_thickness_of_precipitation_amount_above_threshold",
-    description="{freq} number of days with precipitation above a daily percentile."
+    description="{freq} number of days with precipitation above percentile."
     " Only days with at least {thresh} are counted.",
     units="days",
     cell_methods="time: sum over days",
     compute=indices.days_over_precip_thresh,
 )
 
+days_over_precip_doy_thresh = Precip(
```
Comment (bzah) [inline xclim/indicators/atmos/_precip.py:300]: I forgot to do the same on `fraction_over_precip_thresh`.

File: xclim/indices/_multivariate.py [hunk 2]
```diff
@@ -1106,7 +1106,8 @@ def days_over_precip_thresh(
     pr : xarray.DataArray
       Mean daily precipitation flux.
     per : xarray.DataArray
-      Daily percentile of wet day precipitation flux.
+      Percentile of wet day precipitation flux. Either computed daily (one value per day
+      of year) or computed over a period (one value per spatial point).
```
Comment (huard) [inline xclim/indices/_multivariate.py:1110]: I guess this should also apply to `fraction_over_precip_thresh`.

**Review by huard — APPROVED** [review]

File: xclim/indicators/atmos/_precip.py [hunk 3]
```diff
@@ -307,7 +316,7 @@ class HrPrecip(Hourly):
 )
 
 
-fraction_over_precip_thresh = Precip(
+fraction_over_precip_doy_thresh = Precip(
     identifier="fraction_over_precip_thresh",
```
Comment (aulemahal) [inline xclim/indicators/atmos/_precip.py:320]: This identifier should be different than the one of `fraction_over_precip_thresh`.

File: xclim/indices/_multivariate.py [hunk 4]
```diff
@@ -1189,6 +1195,10 @@ def fraction_over_precip_thresh(
     if "dayofyear" in per.coords:
         # Create time series out of doy values.
         tp = resample_doy(tp, pr)
+    else:
```
Comment (aulemahal) [inline xclim/indices/_multivariate.py:1198]: Shouldn't it be `elif bootstrap:` ?

**Review by aulemahal — COMMENTED** [review]
Sorry, I should have looked into this with more care before the merge. There's 2 issues I see. The first is triggering warnings each time xclim is loaded. The second I have not tested.

File: xclim/indices/_multivariate.py (hunk 4 above)
Comment (bzah) [reply xclim/indices/_multivariate.py:1198]: Yes it should, oopsi

**Pull request merged**